app/components/onboarding_telegram_form/onboarding_telegram_form.html.erb

```diff
@@ -1,27 +1,35 @@
-<%= form_with(
-  model: contributor,
-  hidden: true,
-  local: true,
-  url: onboarding_telegram_path,
-  class: class_attr,
-) do %>
-  <h1 class="OnboardingTelegramForm-header"><%= t 'components.onboarding_telegram_form.heading'%></h1>
-  <p class="OnboardingTelegramForm-subheader"><%= t 'components.onboarding_telegram_form.text'%></p>
-  <%= c 'field', object: contributor, id: :first_name do |field| %>
-    <%= c 'input', field.input_defaults.merge(value: first_name, required: true) %>
-  <% end %>
+<%= c 'stack', **attrs do %>
+
+  <%= c 'stack', class: 'text-serif', space: :xsmall do %>
+    <%= c 'heading', tag: :h1 do %>
+      <%= t 'components.onboarding_telegram_form.heading' %>
+    <% end %>
 
-  <%= c 'field', object: contributor, id: :last_name do |field| %>
-    <%= c 'input', field.input_defaults.merge(value: last_name, required: true) %>
+    <p><%= t 'components.onboarding_telegram_form.text' %></p>
   <% end %>
 
-  <%= c 'field', object: @contributor, id: :data_processing_consent, styles: [:horizontal, :leftAligned] do |field| %>
-    <%= c 'checkbox', field.input_defaults.merge(required: true, value: nil) %>
+  <%= form_with(model: contributor, local: true, url: onboarding_telegram_path) do %>
+    <%= c 'stack' do %>
+      <%= c 'field', object: contributor, id: :first_name do |field| %>
+        <%= c 'input', field.input_defaults.merge(required: true) %>
+      <% end %>
+
+      <%= c 'field', object: contributor, id: :last_name do |field| %>
+        <%= c 'input', field.input_defaults.merge(required: true) %>
+      <% end %>
+
+      <%= c 'input', id: 'contributor[telegram_onboarding_token]', type: 'hidden', value: contributor.telegram_onboarding_token %>
```

Comment:
Any specific reason to set the Telegram token in `#show`, potentially allowing a contributor to alter it? Instead of setting it on the fly in `#create` before persisting the record to the database?

I don’t really see if/how this would be a problem, but it seems a little unusual.

Reply:
Yes, makes sense. I can overwrite `contributor_params` and `merge(telegram_onboarding_token: Securerandom..)`

Reply:
No, I decided against this. See 5ac3f6101a2e57e399d86720ce5982bc7c277237